modules/andes-core/broker/src/main/java/org/dna/mqtt/wso2/AndesMQTTBridge.java

```diff
@@ -239,4 +242,39 @@ public void onProcessPingRequest(String clientID){
         }
         MQTTopicManager.getInstance().processPingRequest(clientID);
     }
+
+
+    /**
+     * Sends retain messages if exist upon MQTT subscription.
+     *
+     * @param topicName topic name of the subscription.
+     * @param clientID subscription client id
+     */
+    public void sendRetainMessageToSubscriber(String topicName, String clientID) {
+
+        // Send retained message if available
+        try {
+            List<DeliverableAndesMetadata> metadataList = Andes.getInstance().getRetainedMetadataByTopic(topicName);
+
+            for (DeliverableAndesMetadata metadata : metadataList) {
+                AndesContent content = Andes.getInstance().getRetainedMessageContent(metadata);
```

Comment:
If we are doing this approach need to change current messaging engine methods for retain to fetch metadata for messageId list. IMO we can do this as a improvement to retain message flow in future.